bokeh/io.py

```diff
@@ -413,6 +414,8 @@ def _get_save_args(state, filename, resources, title):
     if filename is None:
         warn = False
         filename = _detect_filename("html")
+        if not os.access(filename, os.W_OK):
```

Comment:
You need `X_OK` (executable bit) on a directory to make it "useful". Try `chmod -x on_some_directory` and then `touch on_some_directory/whatever` and see what happens.